internal/gateways/vehicle_signal_decoding_service.go

```diff
@@ -0,0 +1,118 @@
+package gateways
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"io"
+	"time"
+
+	"github.com/DIMO-Network/shared"
+)
+
+var ErrNotFound = errors.New("not found")
+var ErrBadRequest = errors.New("bad request")
+
+//go:generate mockgen -source vehicle_signal_decoding_service.go -destination mocks/vehicle_signal_decoding_service_mock.go
+type VehicleSignalDecodingAPIService interface {
+	GetPIDsTemplateByVIN(vin string) (*PIDConfigResponse, error)
+	GetUrls(vin string) (*UrlConfigResponse, error)
+}
+
+type UrlConfigResponse struct {
+	PidURL           string `json:"pidUrl"`
+	DeviceSettingURL string `json:"deviceSettingUrl"`
+	DbcURL           string `json:"dbcURL"`
+	Version          string `json:"version"`
+}
+
+type PIDConfigResponse struct {
+	Requests     []PIDConfigItemResponse `json:"requests"`
+	TemplateName string                  `json:"template_name"`
+	Version      string                  `json:"version"`
+}
+
+type PIDConfigItemResponse struct {
+	ID              int64  `json:"id"`
+	Header          int64  `json:"header"`
+	Mode            int64  `json:"mode"`
+	Pid             int64  `json:"pid"`
+	Formula         string `json:"formula"`
+	Protocol        string `json:"protocol"`
+	IntervalSeconds int    `json:"interval_seconds"`
+	Name            int64  `json:"name"`
+	Version         string `json:"version"`
+}
+
+type vehicleSignalDecodingAPIService struct {
+	httpClient shared.HTTPClientWrapper
+}
+
+const VEHICLE_SIGNAL_DECODING_API_URL = "https://vehicle-signal-decoding.dev.dimo.zone"
+
+func NewVehicleSignalDecodingAPIService() VehicleSignalDecodingAPIService {
+	h := map[string]string{}
+	hcw, _ := shared.NewHTTPClientWrapper(VEHICLE_SIGNAL_DECODING_API_URL, "", 10*time.Second, h, true) // ok to ignore err since only used for tor check
+
+	return &vehicleSignalDecodingAPIService{
+		httpClient: hcw,
+	}
+}
+
+func (v *vehicleSignalDecodingAPIService) GetPIDsTemplateByVIN(vin string) (*PIDConfigResponse, error) {
+	res, err := v.httpClient.ExecuteRequest(fmt.Sprintf("/v1/device-config/%s/pids", vin), "GET", nil)
```

Comment:
then change this to only be pid Url eg

```suggestion
	res, err := v.httpClient.ExecuteRequest(pidURL, "GET", nil)
```